### PR TITLE
chore: fixes for leanprover/lean4#2734

### PR DESCRIPTION
**Pull request by semorrison**
This is stacked on top of #297.

### CODE REVIEW

File: Std/Tactic/Ext/Attr.lean [hunk 1]
```diff
@@ -20,31 +20,35 @@ structure ExtTheorem where
...
 /-- The state of the `ext` extension environment -/
 structure ExtTheorems where
   /-- The tree of `ext` extensions. -/
-  tree   : DiscrTree ExtTheorem true := {}
+  tree   : DiscrTree ExtTheorem := {}
   /-- Erased `ext`s. -/
   erased  : PHashSet Name := {}
   deriving Inhabited
 
+
```
Comment (digama0) [inline Std/Tactic/Ext/Attr.lean:34]:
```suggestion
```

**Review by joehendrix — APPROVED** [review]
The changes look good to me.

File: Std/Tactic/Relation/Rfl.lean [hunk 2]
```diff
@@ -17,11 +17,14 @@ namespace Std.Tactic
 
 open Lean Meta
 
+/-- Discrimation tree settings for the `refl` extension. -/
+def reflExt.config : WhnfCoreConfig := {}
+
 /-- Environment extensions for `refl` lemmas -/
 initialize reflExt :
-    SimpleScopedEnvExtension (Name × Array (DiscrTree.Key true)) (DiscrTree Name true) ←
+    SimpleScopedEnvExtension (Name × Array (DiscrTree.Key)) (DiscrTree Name) ←
```
Comment (digama0) [inline Std/Tactic/Relation/Rfl.lean:25]:
```suggestion
    SimpleScopedEnvExtension (Name × Array DiscrTree.Key) (DiscrTree Name) ←
```

File: Std/Tactic/Relation/Symm.lean [hunk 3]
```diff
@@ -18,11 +18,14 @@ open Lean Meta
 
 namespace Std.Tactic
 
+/-- Discrimation tree settings for the `symm` extension. -/
+def symmExt.config : WhnfCoreConfig := {}
+
 /-- Environment extensions for symm lemmas -/
 initialize symmExt :
-    SimpleScopedEnvExtension (Name × Array (DiscrTree.Key true)) (DiscrTree Name true) ←
+    SimpleScopedEnvExtension (Name × Array (DiscrTree.Key)) (DiscrTree Name) ←
```
Comment (digama0) [inline Std/Tactic/Relation/Symm.lean:26]:
```suggestion
    SimpleScopedEnvExtension (Name × Array DiscrTree.Key) (DiscrTree Name) ←
```

**Review by digama0 — APPROVED** [review]

**Pull request closed without merging**